crates/sncast/src/lib.rs

```diff
@@ -384,16 +382,19 @@ pub async fn handle_wait_for_tx<T>(
         .await
         {
             Ok(_) => Ok(return_value),
-            Err(message) => Err(anyhow!(message)),
+            Err(message) => {
+                eprintln!("Transaction hash: {transaction_hash:#x}\n");
+                Err(anyhow!(message))
+            }
         };
     }
 
     Ok(return_value)
 }
 
-pub fn print_formatted(output: Vec<(&str, String)>, json: bool, error: bool) -> Result<()> {
+pub fn print_formatted(output: Vec<(String, String)>, json: bool, error: bool) -> Result<()> {
```

Comment:
@MaksymilianDemitraszek can you please take a look at this in your free time? Perhaps Olek can somehow adjust to your PR?

Reply:
This PR should be easily adaptable to changes I made, my changes should allow reuse more logic.

Reply:
not relevant anymore